backend/inspirehep/disambiguation/tasks.py

```diff
@@ -70,26 +72,97 @@ def disambiguate_signatures(self, clusters):
     db.session.commit()
 
 
+def match_literature_author(author_data, matcher_config):
+    matched_records = (
+        matched_record["inner_hits"]["authors"]["hits"]["hits"]
+        for matched_record in match(author_data, matcher_config)
+    )
+    matched_authors_refs = []
+    for record_authors in matched_records:
+        matched_authors_refs.extend(
+            author["_source"]["record"]["$ref"]
+            for author in record_authors
+            if "record" in author["_source"]
+        )
+    if len(set(matched_authors_refs)) == 1:
+        author_reference_url = matched_authors_refs[0]
+        return {"$ref": author_reference_url}
+
+
+def assign_reference_to_author(author, updated_authors, reference):
+    author["record"] = reference
+    updated_authors.append(reference["$ref"].split("/")[-1])
```

Comment:
too much mutation makes things hard to understand. I think it's better to
```suggestion
    return reference["$ref"].split("/")[-1]
```
and accumulate stuff outside. This also avoids having to pass `updated_authors` down to functions which don't really need it to do their job.